docs/src/pages/components/data-grid/filtering/CustomFilterPanelContent.tsx

```diff
@@ -0,0 +1,91 @@
+import * as React from 'react';
+import { DataGridPro, GridLinkOperator, GridToolbar } from '@mui/x-data-grid-pro';
+import { useDemoData } from '@mui/x-data-grid-generator';
+
+const VISIBLE_FIELDS = ['name', 'rating', 'country', 'dateCreated', 'isAdmin'];
+
+const initialState = {
+  filter: {
+    filterModel: {
+      items: [
+        {
+          id: 1,
+          columnField: 'name',
+          operatorValue: 'contains',
+          value: 'D',
+        },
+        {
+          id: 2,
+          columnField: 'name',
+          operatorValue: 'contains',
+          value: 'D',
+        },
+        {
+          id: 3,
+          columnField: 'rating',
+          operatorValue: '>',
+          value: '0',
+        },
+      ],
+    },
+  },
+};
+
+export default function CustomFilterPanelContent() {
+  const { data } = useDemoData({
+    dataSet: 'Employee',
+    visibleFields: VISIBLE_FIELDS,
+    rowLength: 100,
+  });
+
+  return (
+    <div style={{ height: 400, width: '100%' }}>
+      <DataGridPro
+        {...data}
+        components={{
+          Toolbar: GridToolbar,
+          // Use custom FilterPanel only for deep modification
+          // FilterPanel: MyCustomFilterPanel,
+        }}
+        componentsProps={{
+          filterPanel: {
+            // Force to use "And" operator
+            linkOperators: [GridLinkOperator.And],
+            // Display columns by ascending alphabetical order
+            columnsSort: 'asc',
+            filterFormProps: {
+              // Customize inputs by passing props
+              columnInputProps: {
+                variant: 'outlined',
+                size: 'small',
+                sx: { mt: 'auto' },
+              },
+              operatorInputProps: {
+                variant: 'outlined',
+                size: 'small',
+                sx: { mt: 'auto' },
+              },
+              valueInputProps: {
+                required: true,
+              },
```

Comment:
Not sure this is the best way to customize the deleteIcon color.
But I think we could add a simple example for deleteIconProps too.

```suggestion
              deleteIconProps: {
                sx: {
                  "& .MuiSvgIcon-root": { color: "red" }
                }
              },
```

Reply:
Yes, it's better than my reorder which was a bad accessibility practice